lib/elixir/lib/calendar/iso.ex

```diff
@@ -32,18 +32,17 @@ defmodule Calendar.ISO do
   @microseconds_per_second 1_000_000
   @parts_per_day @seconds_per_day * @microseconds_per_second
 
+  @sep [?\s, ?T]
   @days_per_nonleap_year 365
   @days_per_leap_year 366
-
   @months_in_year 12
 
   # The ISO epoch starts, in this implementation,
   # with ~D[0000-01-01]. Era "1" starts
   # on ~D[0001-01-01] which is 366 days later.
   @iso_epoch 366
 
-  @doc false
-  def __match_date__ do
+  [match_date, guard_date, read_date] =
```

Comment:
Would you consider making these date and time match/guard clauses part of the public API? I appreciate they are intended to support ISO8601 formats only but for a large class of business calendars that are also based upon the Gregorian calendar the date/time/datetime formats are the same - only the calendar is different from a parsing perspective. It would be great to leverage a consistent way to decompose the date/time/datetime strings.

Reply:
For other calendars, I would perhaps consider using `nimble_parsec`? That's literally from where I got this code from, except we can't depend on it from Elixir itself. :(

Reply:
Definitely doable. My thinking was more that its such a common pattern for derivative Gregorian calendars that it would be DRY to reuse it. But I can understand the counter perspective as well.